git/Cheatsheat.md

```diff
@@ -0,0 +1,100 @@
+- [How to rebase properly?](https://github.com/lambdaworkshub/contributing-guide/blob/master/git/Cheatsheat.md#how-to-rebase-properly)
+
+### Writing Good Commit Message
+
+Provided [Erlang's guide](https://github.com/erlang/otp/wiki/writing-good-commit-messages) gives rules for writing good commit messages.
+
+- Write your commits in **imperative** style.
+
+    **GOOD:** `Remove unused code`
+
+    **WRONG:** ~~`Removed unused code`~~
+
+- Don't put **punctuation marks** at the end of commit message.
+
+    **GOOD:** `Remove unused code`
+
+    **WRONG:** ~~`Remove unused code.`~~
+
+- Use short commit messages - up to 50 characters.
+
+    **GOOD:** `Remove unused code`
+
+    **WRONG:** ~~`In this commit we've removed unused code along with classes`~~
+
+- Don't put more statements in one commit message. Use description instead. Description should be wrapped to 80 characters maximum.
+
+    **GOOD:**
+    ```
+    Remove unused code
+
+    Removed 'getAllUsers' function from 'UserDAO' class.
+    Performed cleanup on 'Users' controller. Removed unused
+    config file.
+    ```
+
+    **WRONG:** ~~`Remove 'getAllUsers' function. Cleanup 'Users' controller. Remove unused config.`~~
+
+- Leave second line blank when writing descriptions.
+
+    **GOOD:**
+    ```
+    Remove unused code
+
+    Removed 'getAllUsers' function from 'UserDAO' class.
+    Performed cleanup on 'Users' controller. Removed unused
+    config file.
+    ```
+
+    **WRONG:**
+    ```
+    Remove unused code
+    Removed 'getAllUsers' function from 'UserDAO' class.
+    Performed cleanup on 'Users' controller. Removed unused
+    config file.
+    ```
+
+### Useful Git commands and how to use them
+
+As it is assumed that basic Git commands are widely known, the focus will move onto less known yet useful commands instead.
+
+1. `git commit --amend`
+    This command is used to avoid creation of two commits with the same or similar purpose. For example, if we push some code and we notice that we forget to delete `print` from our code, then there's no sense to make another commit with `Remove unused 'print' command`. Then, just remove it with use of `--amend` flag instead.
+2. `git reset HEAD~`
+    This command is used to **undo** last commit locally. If you type `git commit -m "Implement something"` and you notice you've made mistake, then you can use this command to reset your latest work.
+
+### How to rebase properly?
+
+Rebasing technique is used when another branch is merged before yours for avoiding `Git default merge` commits during **resolving conflicts**. Instead, it's recommended to perform process to **put** your branch on updated `dev` instead.
+
+Assume there are two branches `branch-1` and `branch-2`:
+
+```
+git checkout -b branch-1 dev
+git checkout -b branch-2 dev
+```
+
+If work is performed on `branch-2`, but `branch-1` is merged to `dev` in the meantime. Then `branch-2` needs to be **rebased**. To do that, type following commands:
```

Comment:
It doesn't need to be rebased, but some Git flows (ours, for example) require so.